pkg/image/apiserver/admission/apis/imagepolicy/v1/register.go

```diff
@@ -44,3 +44,24 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 }
 
 func (obj *ImagePolicyConfig) GetObjectKind() schema.ObjectKind { return &obj.TypeMeta }
+
+var GroupVersion = schema.GroupVersion{Group: "image.openshift.io", Version: "v1"}
```

Comment:
not sure i'm sold on the group name?

admission.openshift.io maybe?

Reply:
> not sure i'm sold on the group name?
> 
> admission.openshift.io maybe?

We are matching the names for the admission plugin.   It is feature focused.  In this case, the feature is related to control over images.  A broad name like "admission", just makes a union bucket without ownership.

Reply:
well i'd expect each admission plugin that needs config to contribute a resource type to that group.  In the same way that we have a config.openshift.io group.

creating a million groups seems like it makes it harder to track down all the admission config types that exist.

and how does this reconcile with the "images" resource that's in "config.openshift.io" which is also feature focused on control over image behavior?

And what if we re-introduce image-import configuration?  Where does that end up going?

Reply:
(ie by doing this you're effectively creating two potential locations for that additional configuration information and it's not obvious to me which one would be "right")

Reply:
ok. I can appreciate the distinction. Still think i'd prefer to see a higher-order group name to bucket "process facing" configuration into.  ie image.$something.openshift.io/ImagePolicy because image.openshift.io feels too flat to me.

no to mention that "image.openshift.io" is already used by the image api itself, so you're comingling user-facing resources and admin-facing process config in the same group.

Reply:
> not to mention that "image.openshift.io" is already used by the image api itself, so you're comingling user-facing resources and admin-facing process config in the same group.

They are both owned by the same feature: images.  I don't see a need add depth to the tree of groups.  The same goes for all the rest, they are logically parented.

Now, in the interest of staying out of what a particular group does, I could live with something like `admission.image.openshift.io`. That positions us well for separating the apiservers running our particular features, but I don't think the depth actually helps users.

Reply:
I think the question is fundamentally whether the hierarchy should be:

0) product (openshift)
1) type of data (process-config/admission config)
2) specific feature/component (images)

or

0) product (openshift)
1) specific feature/component (images)
2) type of data (process-config/admission config)

I can live w/ the latter (what you seem to be suggesting as a compromise) but it's probably worth soliciting a few more opinions since this won't be the last new api group we create.  Our existing hierarchies could be interpreted either way (ie is "config" the component, or the type of data, in "config.openshift.io")

Reply:
> I can live w/ the latter (what you seem to be suggesting as a compromise) but it's probably worth soliciting a few more opinions since this won't be the last new api group we create. Our existing hierarchies could be interpreted either way (ie is "config" the component, or the type of data, in "config.openshift.io")

Dividing our produce by end feature (`image.openshift.io`, not `admission.openshift.io`) makes more sense than dividing by the way the implementation works.  We wouldn't create a `image.controller.openshift.io`.  Dividing based on method of implementation makes a lot of sense to people who are implementing the work, but doesn't make a lot of sense to people using it.  As we try to become more extensible, I suspect images will become indistinguishable from other external packages we use and embed.